sockeye/train.py

```diff
@@ -92,6 +91,16 @@ def check_arg_compatibility(args: argparse.Namespace):
     check_condition(args.optimized_metric == C.BLEU or args.optimized_metric in args.metrics,
                     "Must optimize either BLEU or one of tracked metrics (--metrics)")
 
+    if args.encoder == C.TRANSFORMER_TYPE:
+        check_condition(args.transformer_model_size == args.num_embed[0],
```

Comment:
If that is the case, could we remove the args for transformer model sizes and just take the corrsponding embedding sizes?

Reply:
not sure, args is fixed and cannot be changed and I don't like these implicit overrides. This fails early and makes it clear what to change.
In the future we can think about an optional projection layer from num_embed to model_size if needed.

Reply:
I was referring to removing the argument altogether, not to overwriting args.

Reply:
I see. I would be in favor of keeping it as it refers to the parameter introduced by Vaswani etal, 17. Overloading --num-embed for something else seems quite complicated. Especially since the other architectures have their own 'size' args, for example `--rnn-num-hidden`.